File: C2/PDOO/Practicas/DeepSpaceJava/src/deepspace/GameStateController.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author Profesor
 */
class GameStateController {
    private GameState state;
    
    GameStateController() {
        state=GameState.CANNOTPLAY;        
    }

    public GameState getState() {
        return state;
    }
    
    public GameState next(int turn,int nPlayers) {
        switch (state) {
            case CANNOTPLAY: 
                state=GameState.INIT;
                break;
            case INIT: 
                state=GameState.AFTERCOMBAT;
                break;                
            case BEFORECOMBAT: 
                state=GameState.AFTERCOMBAT;
                break;
            case AFTERCOMBAT: 
                if (turn>=nPlayers)
                    state=GameState.BEFORECOMBAT;
                else
                    state=GameState.INIT;
                break;                
        }
        return state;
    }
}

File: C2/PDOO/Practicas/DeepSpaceRuby/lib/GameStateController.rb
#encoding:utf-8

require_relative 'GameState'

module Deepspace

# 7.4.17 - Translation from Java
# @author Profesor
    
class GameStateController 

    attr_reader :state
    
    def initialize () 
        @state=GameState::CANNOTPLAY;        
    end

    
    def next(turn,nPlayers) 
        case @state
            when GameState::CANNOTPLAY
                @state=GameState::INIT
            when GameState::INIT
                @state=GameState::AFTERCOMBAT
            when GameState::BEFORECOMBAT
                @state=GameState::AFTERCOMBAT
            when GameState::AFTERCOMBAT
                if (turn>=nPlayers)
                    @state=GameState::BEFORECOMBAT
                else
                    @state=GameState::INIT
                end
        end
        return @state;
    end
end # class

end # module

if $0 == __FILE__ then
  gsc = Deepspace::GameStateController.new()
  puts gsc.state
  20.times {
    i = rand(5)+1
    puts "Turno " + i.to_s + " de 5 jugadores"
    puts gsc.next(i,5)
  }
end
